From 40f74417c434e5dfa9d0ed76d149c0317b45ff53 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Thu, 19 Dec 2019 00:28:05 +0200
Subject: [PATCH 047/269] s32g: Remove VIRTUAL_PLATFORM definition

Old macro VIRTUAL_PLATFORM was still being defined. One side-effect of
that was that the system timer frequency (cntfrq_el0) was being set with
the value from the VDK simulator, resulting in impossibly fast
countdowns in U-Boot.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-4497
---
 lib/el3_runtime/aarch64/context_mgmt.c |  2 --
 plat/s32g/include/platform_def.h       | 16 ++++------------
 plat/s32g/platform.mk                  |  3 ---
 3 files changed, 4 insertions(+), 17 deletions(-)

diff --git a/lib/el3_runtime/aarch64/context_mgmt.c b/lib/el3_runtime/aarch64/context_mgmt.c
index 77afe9a8c..e0e429849 100644
--- a/lib/el3_runtime/aarch64/context_mgmt.c
+++ b/lib/el3_runtime/aarch64/context_mgmt.c
@@ -75,9 +75,7 @@ void cm_setup_context(cpu_context_t *ctx, const entry_point_info_t *ep)
 	security_state = GET_SECURITY_STATE(ep->h.attr);
 
 	/* Clear any residual register values from the context */
-#ifndef S32G_VIRTUAL_PLATFORM
 	zeromem(ctx, sizeof(*ctx));
-#endif
 
 	/*
 	 * SCR_EL3 was initialised during reset sequence in macro
diff --git a/plat/s32g/include/platform_def.h b/plat/s32g/include/platform_def.h
index b632e9990..04b871849 100644
--- a/plat/s32g/include/platform_def.h
+++ b/plat/s32g/include/platform_def.h
@@ -54,20 +54,12 @@
 #define PLAT_MAX_PWR_LVL_STATES		2
 
 #define PLAT_PRIMARY_CPU		0x0
-/* Generic timer frequency; this goes directly into CNTFRQ_EL0 */
-#if defined S32G_VIRTUAL_PLATFORM
-/* Dummy value, to make time passing more palatable on the functional sim.
- * We suspect the sim incorrectly further divides the clock signal, so we
- * manually (and approximately) adjust for that.
- */
-#define COUNTER_FREQUENCY	0xC0000
-#else
-/* 5MHz; this is based on the assumption that GPR00[CA53_COUNTER_CLK_DIV_VAL]
- * contains the reset value of 0x7, hence producing a divider value of 8,
- * applied to the FXOSC frequency of 40MHz
+/* Generic timer frequency; this goes directly into CNTFRQ_EL0.
+ * Its end-value is 5MHz; this is based on the assumption that
+ * GPR00[CA53_COUNTER_CLK_DIV_VAL] contains the reset value of 0x7, hence
+ * producing a divider value of 8, applied to the FXOSC frequency of 40MHz.
  */
 #define COUNTER_FREQUENCY	0x004C4B40
-#endif
 
 #define SIUL2_0_BASE_ADDR	0x4009C000UL
 #define SIUL2_1_BASE_ADDR	0x44010000UL
diff --git a/plat/s32g/platform.mk b/plat/s32g/platform.mk
index 207007919..37304c8d4 100644
--- a/plat/s32g/platform.mk
+++ b/plat/s32g/platform.mk
@@ -72,6 +72,3 @@ CFLAGS			+= -O0
 CRASH_REPORTING		:= 1
 # As verbose as it can be
 LOG_LEVEL		:= 50
-# Enable simulator-specific workarounds. To be removed after hw is available.
-S32G_VIRTUAL_PLATFORM	:= 1
-$(eval $(call add_define_val,S32G_VIRTUAL_PLATFORM,$(S32G_VIRTUAL_PLATFORM)))
-- 
2.17.1

